### ISSUE TITLE
Investigate using snippets for autocompleting the available functions in all_functions.rs

### ISSUE THREAD

**Issue by JoshuaBatty** [nannou-org/nannou#236]
just reading over the all_functions.rs example. Just thinking from a usability perspective, would be awesome to enable auto complete snippets for these functions. 

Say I wanted to access the mouse_wheel... instead of going back to the all_functions.rs example to copy and paste the function signature I could just start typing `fn mouse_whe` and that should be enough for the IDE to know you want access to the mouse_wheel function an auto fill out the function below.

```
fn mouse_wheel(_app: &App, _model: &mut Model, _dt: MouseScrollDelta, _phase: TouchPhase) {
}
```

Might only be able to support VS_Code officially for something like this, but haven't looked too hard yet. Just a thought for now.

**Comment by mitchmindtree** [comment]
Yeah fully, would be awesome if someone reading this who happened to be familiar with VSCode/Vim/IDE-of-your-choice could put some smart completion thing together like this :) We'd be happy to host it under nannou-org!

**Comment by JoshuaBatty** [comment]
I might look at what would be involved in setting this up for VSCode. I can imagine it would be awesome if there was a Nannou plugin that you could install directly in VSCode itself. Could contain these handy snippets, hot keys for auto compiling etc...
